.github/workflows/tests.yml

```diff
@@ -43,7 +43,7 @@ jobs:
             python-version: "3.9"
 
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
```

Comment:
use commit hashes instead?

Reply:
We are adding dependabot here to handle that automatically.

Reply:
AFAIK dependabot will replace tags with tags and commits hashes with commits hashes

Reply:
Oh, really? That's unfortunate 😬 Let's see if there's a tool to automatically convert.

Reply:
Used https://github.com/mheap/pin-github-action and some post-processing. All pinned now.

Reply:
Do we want to add `<major>.<minor>.<patch>` version numbers to all of them to make it easier to see which version is used exactly?

Reply:
Ah, I see why now. Cool. Added!